python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/router_ospf.py

```diff
@@ -31,7 +31,5 @@ def router_ospf(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> Non
                 no_passive_interfaces.append(p2p_link_data["interface"])
 
         if no_passive_interfaces:
-            self.structured_config.router_ospf.process_ids.append_new(
-                id=self.inputs.underlay_ospf_process_id,
-                no_passive_interfaces=no_passive_interfaces,
-            )
+            process = self.structured_config.router_ospf.process_ids.obtain(self.inputs.underlay_ospf_process_id)
```

Comment:
This could potentially leave duplicate interfaces in this list, but I guess we will detect that somewhere else? I think we should obtain it at the top and just append_unique directly in the loop instead of collecting them.

```suggestion
            # TODO: Move this to the top and append_unique directly in the loop above.
            process = self.structured_config.router_ospf.process_ids.obtain(self.inputs.underlay_ospf_process_id)
```